docs/data/data-grid/performance/GridWithReactMemo.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { unstable_useForkRef as useForkRef } from '@mui/utils';
+import {
+  DataGridPro,
+  GridRow,
+  DataGridProColumnHeaders,
+} from '@mui/x-data-grid-pro';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const TraceUpdates = React.forwardRef<any, any>((props, ref) => {
+  const { Component, ...other } = props;
+  const rootRef = React.useRef<HTMLElement>();
+  const handleRef = useForkRef(rootRef, ref);
+
+  React.useEffect(() => {
+    rootRef.current?.classList.add('updating');
+
+    const timer = setTimeout(() => {
+      rootRef.current?.classList.remove('updating');
+    }, 500);
+
+    return () => clearTimeout(timer);
+  });
+
+  return <Component ref={handleRef} {...other} />;
+});
+
+const RowWithTracer = React.forwardRef((props, ref) => {
+  return <TraceUpdates ref={ref} Component={GridRow} {...props} />;
+});
+
+const ColumnHeadersWithTracer = React.forwardRef((props, ref) => {
+  return <TraceUpdates ref={ref} Component={DataGridProColumnHeaders} {...props} />;
+});
+
+const MemoizedRow = React.memo(RowWithTracer);
```

Comment:
Is it expected that the whole row is rerendered on cell focus change?


https://user-images.githubusercontent.com/13808724/220181308-37c58ac8-290c-4d7e-a19d-09e51f99f39b.mov